src/input/readers/ascii/Ascii.cc

```diff
@@ -345,7 +344,8 @@ bool Ascii::DoUpdate() {
                 FailWarn(fail_on_invalid_lines,
                          Fmt("Not enough fields in line '%s' of %s. Found "
                              "%d fields, want positions %d and %d",
-                             line.c_str(), fname.c_str(), pos, fit.position, fit.secondary_position));
+                             line.c_str(), fname.c_str(), pos, fit.position, fit.secondary_position),
+                         ! fail_on_invalid_lines);
```

Comment:
The `! fail_on_invalid_lines` doesn't make a difference, so passing `true` seems less cryptic to me. What do you think?

Reply:
That decision was to avoid the case where failing on invalid lines would cause an error, then later a completely unrelated error gets suppressed. Since it aborts immediately it shouldn't make a difference for this function, but it may make a difference to the caller, theoretically.

Though, admittedly no where else seems to do anything like that, so it could just get changed back. I thought it would be nice to have conditionally (and do that sort of thing more).

As a side note, the backend just suppresses and unsuppresses warnings unconditionally, it would be nice to have some memory of `suppress_warnings` and reset it - so calling `DoUpdate()` won't always stop suppression. This is just one tiny case of trying to be a little more clever with suppression logic, but it's also not really necessary